cli.go

```diff
@@ -163,6 +182,7 @@ func (cli *CLI) Run(args []string) int {
 				if err != nil {
 					return logError(err, ExitCodeRunnerError)
 				}
+				runner.SetReadyChannel(readyCh)
```

Comment:
Given that we're shutting down the runners in these signal handlers, shouldn't we have a context on the ready channel that we can close to stop that goroutine as well?

Reply:
Hello @tgross 
Thank you very much for your contribution and time spent!
If I understood your request correctly, then the idea is to create a context with cancellation, pass the context to the goroutine listening to the channel, and сanceled the context when shutting down the runners.


I apologize if this implementation isn't perfect, as this is my first experience making such changes. Please review the changes and let me know if further adjustments are needed.

Reply:
As for the fallen CI pipeline with tests, I noticed that there is a similar problem for neighboring PRs. I will be glad to help you fix the dropped tests

Reply:
> I apologize if this implementation isn't perfect, as this is my first experience making such changes. Please review the changes and let me know if further adjustments are needed.

Looks very close. I've added a suggestion for two one-line fixes and this will be good-to-merge.

> As for the fallen CI pipeline with tests, I noticed that there is a similar problem for neighboring PRs. I will be glad to help you fix the dropped tests

Oof, ok... let's not worry about it in this PR then.